src/java.base/share/classes/jdk/internal/loader/ClassLoaders.java

```diff
@@ -197,10 +197,11 @@ protected PermissionCollection getPermissions(CodeSource cs) {
 
         /**
          * Called by the VM to support dynamic additions to the class path
+         * Locks the System ClassLoader to prevent concurrent changes.
          *
          * @see java.lang.instrument.Instrumentation#appendToSystemClassLoaderSearch
          */
-        void appendToClassPathForInstrumentation(String path) {
+        synchronized void appendToClassPathForInstrumentation(String path) {
```

Comment:
We might not need this.  appendClasspath is thread safe so it's okay for several agents calling appendToSystemClassLoaderSearch at around the same time. I don't think it needs to sycnrhonize with anything else.

Reply:
I traced it down to this:  Is this why it's already synchronized ?
  appendClassPath -> ucp.addFile -> addFile -> addURL
    public synchronized void addURL(URL url) {
        if (closed || url == null)
            return;
        synchronized (unopenedUrls) {
            if (! path.contains(url)) {
                unopenedUrls.addLast(url);
                path.add(url);
            }
        }
    }

Reply:
Yes, URLClassPath is already synchronized. I'm not 100% sure why the ObjectLocker is there but it's possible to pre-dates parallel capable class loaders so it dates from a time when findClass/loadClass were synchronized on "this".

Reply:
Yes, the code's been there for forever.  Thanks!